sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -347,6 +348,16 @@ public Context setContext(Object[] swaggerMethodArguments) {
         return (context != null) ? context : Context.NONE;
     }
 
+    /**
+     * Get the {@link RequestOptions} passed into the proxy method.
+     *
+     * @param swaggerMethodArguments the arguments passed to the proxy method
+     * @return the request options
+     */
+    public RequestOptions setRequestOptions(Object[] swaggerMethodArguments) {
+        return CoreUtils.findFirstOfType(swaggerMethodArguments, RequestOptions.class);
```

Comment:
This is an ancillary performance question, does the Java proxy interface pass the Object[] in a consistent order? I'm wondering if there could be an optimization for this.

Reply:
AutoRest generated code does, but I'm not sure if azure-core could make the assumption that all code follows the same order.

However, a `CoreUtils.findLastOfType()` might slightly improve the perf.